File: src_local/agents/Support/discovery_agent.ts
import winston from "winston";

const logger = winston.child({ module: "DiscoveryAgent" });

export class DiscoveryAgent {
    orc: any;

    constructor(orchestrator: any) {
        this.orc = orchestrator;
    }

    async runDiscoveryPhase(): Promise<[any, any[]]> {
        logger.info("🔭 Iniciando fase de descoberta...");

        const ctx = await this.orc.contextEngine.analyzeProject();
        let findings = await this.orc.runStrategicAudit(ctx, null, false);

        const obfuscationFindings = await this.orc.runObfuscationScan();
        findings = [...findings, ...obfuscationFindings];

        return [ctx, findings];
    }
}

File: src_local/agents/Support/discovery_agent.py
import logging

logger = logging.getLogger(__name__)

class DiscoveryAgent:
    """Agente especialista em descoberta topológica e escaneamento de segurança."""
    
    def __init__(self, orchestrator):
        self.orc = orchestrator

    def run_discovery_phase(self):
        """Executa Auditoria Estratégica e Scan de Ofuscação."""
        ctx = self.orc.context_engine.analyze_project()
        findings = self.orc.run_strategic_audit(ctx, include_history=False)
        findings.extend(self.orc._run_obfuscation_scan())
        return ctx, findings
